e2e/test/scenarios/question/reproductions/22230-query-builder-aggregation-filtering.cy.spec.js

```diff
@@ -0,0 +1,43 @@
+import {
+  restore,
+  popover,
+  summarize,
+  openPeopleTable,
+  visualize,
+} from "e2e/support/helpers";
+
+describe("issue 22230", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("should allow filtering on aggregated fields", () => {
+    openPeopleTable({ mode: "notebook" });
+
+    summarize({ mode: "notebook" });
+    popover().findByRole("option", { name: "Maximum of ..." }).click();
+    // I don't know why this can't be chained, I tried it and it didn't work.
+    popover().findByRole("option", { name: "Name" }).click();
+
+    // findByRole doesn't work here, and I don't know why.
+    cy.findByText("Pick a column to group by").click();
+    popover().findByRole("option", { name: "Source" }).click();
+
+    // can't use `filter()` because it found multiple filter buttons and Cypress only allows clicking on one element at a time.
+    cy.button("Filter").click();
```

Comment:
`filter({ mode: "notebook" })` doesn't work since it found 2 filter button
<img width="350" alt="image" src="https://user-images.githubusercontent.com/1937582/227637127-dc45d6e2-3d50-4670-9c7a-dc579247b2b8.png">